pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -182,13 +183,36 @@ func (p *Provider) loadConfigurationFromIngresses(ctx context.Context, client Cl
 		TCP: &dynamic.TCPConfiguration{},
 	}
 
+	major, minor, err := client.GetServerVersion()
+	if err != nil {
+		log.FromContext(ctx).Errorf("Failed to get server version: %w", err)
+		return conf
+	}
+
+	var ingressClass *networkingv1beta1.IngressClass
+
+	if major >= 1 && minor >= 18 {
+		ic, exists, err := client.GetIngressClass(p.IngressClass)
```

Comment:
maybe we could log an error if no IngressClass is set? Otherwise, the provider will just be empty without a real clue why